server/src/main/java/org/opensearch/search/backpressure/settings/SearchShardTaskSettings.java

```diff
@@ -142,19 +176,45 @@ private void setHeapVarianceThreshold(double heapVarianceThreshold) {
         this.heapVarianceThreshold = heapVarianceThreshold;
     }
 
-    public long getCpuTimeThreshold() {
-        return cpuTimeThreshold;
+    public int getHeapMovingAverageWindowSize() {
+        return heapMovingAverageWindowSize;
+    }
+
+    public void setHeapMovingAverageWindowSize(int heapMovingAverageWindowSize) {
+        this.heapMovingAverageWindowSize = heapMovingAverageWindowSize;
+
+        List<Exception> exceptions = new ArrayList<>();
+        for (Listener listener : listeners) {
+            try {
+                listener.onHeapMovingAverageWindowSizeChanged(heapMovingAverageWindowSize);
+            } catch (Exception e) {
+                exceptions.add(e);
+            }
+        }
+        ExceptionsHelper.maybeThrowRuntimeAndSuppress(exceptions);
+    }
+
```

Comment:
I think its better to break down settings to respective class else this class continues to become a monolith

Reply:
Updated!